@here/harp-mapview/lib/Tile.ts

```diff
@@ -1275,6 +1276,37 @@ export class Tile implements CachedResource {
                     materials[techniqueIndex] = material;
                 }
 
+                // Modify the standard textured shader to support height-based coloring.
+                if (
+                    isTerrainTechnique(technique) &&
+                    technique.heightBasedColors !== undefined &&
+                    technique.displacementMap !== undefined
+                ) {
+                    (material as any).onBeforeCompile = (shader: THREE.Shader) => {
```

Comment:
It is really ugly that we have the custom shader code in the (supposed to be generic) Tile. Ideally there should be a onMaterialCreated callback in the technique so that we can move the code there (or eventually the technique should create the material). Anyways this is out of scope for this change. Will create a ticket for it.